suite-native/module-home/src/components/PortfolioGraphHeader.tsx

```diff
@@ -37,81 +45,109 @@ export const writeOnlyReferencePointAtom = atom<null, ExtendedGraphPoint>(
     },
 );
 
-const percentageDiff = (a: number, b: number) => {
-    if (a === 0 || b === 0) return 0;
-    return 100 * ((b - a) / ((b + a) / 2));
-};
-
 const percentageChangeAtom = atom(get => {
     const selectedPoint = get(selectedPointAtom);
     const referencePoint = get(referencePointAtom);
     return percentageDiff(referencePoint.value, selectedPoint.value);
 });
 
-const Balance = () => {
-    const { FiatAmountFormatter } = useFormatters();
-    const [point] = useAtom(selectedPointAtom);
-
-    return <FiatAmountFormatter value={point.value} />;
-};
-
-const getColorForPercentageChange = (percentageChange: number) =>
-    percentageChange >= 0 ? 'forest' : 'red';
+const headerStyle = prepareNativeStyle(utils => ({
+    marginBottom: utils.spacings.small / 2,
+    color: utils.colors.gray600,
+}));
 
-const PercentageChange = () => {
+const PercentageChange = ({ hasPriceIncreased }: { hasPriceIncreased: boolean }) => {
     const [percentageChange] = useAtom(percentageChangeAtom);
 
     return (
-        <Text color={getColorForPercentageChange(percentageChange)} variant="hint">
+        <Text color={getColorForPercentageChange(hasPriceIncreased)} variant="hint">
             {percentageChange.toFixed(2)}%
         </Text>
     );
 };
 
-const PercentageChangeArrow = () => {
+const PercentageChangeArrow = ({ hasPriceIncreased }: { hasPriceIncreased: boolean }) => {
     const [percentageChange] = useAtom(percentageChangeAtom);
 
     const iconName: IconName = percentageChange >= 0 ? 'arrowUp' : 'arrowDown';
 
     return (
         <Icon
             name={iconName}
-            color={getColorForPercentageChange(percentageChange)}
+            color={getColorForPercentageChange(hasPriceIncreased)}
             size="extraSmall"
         />
     );
 };
 
-const Time = () => {
+const arrowStyle = prepareNativeStyle(() => ({
+    marginRight: 4,
+}));
+
+const priceIncreaseWrapperStyle = prepareNativeStyle<{ hasPriceIncreased: boolean }>(
+    (utils, { hasPriceIncreased }) => ({
+        backgroundColor: utils.transparentize(
+            0.7,
+            hasPriceIncreased ? utils.colors.green : utils.colors.red,
+        ),
+        flexDirection: 'row',
+        alignItems: 'center',
+        paddingHorizontal: utils.spacings.small,
+        paddingVertical: utils.spacings.small / 4,
+        borderRadius: utils.borders.radii.round,
+    }),
+);
+
+const PriceIncreaseIndicator = () => {
+    const { applyStyle } = useNativeStyles();
+    const [percentageChange] = useAtom(percentageChangeAtom);
+    const hasPriceIncreased = percentageChange >= 0;
+
+    return (
+        <Box style={applyStyle(priceIncreaseWrapperStyle, { hasPriceIncreased })}>
```

Comment:
Do we need this style? Did not seen it in Figma and I am quite afraid it will cause perf issue because before only the smallest possible parts (only text) was rerendered, but now you are rerensering quite huge component.

 In case we need it at least create derived atom that will return only true/false because that should minimalize rerenders.

Reply:
[4844d41](https://github.com/trezor/trezor-suite/pull/6778/commits/4844d41ae9331da2dc158882fdc813a878d88138)

As discussed, I created new atom which will only trigger rerender when the price has changed from below zero percent change to above zero and vice versa.